js/apps/system/_admin/aardvark/APP/react/src/views/collections/addCollection/AddCollectionModal.tsx

```diff
@@ -0,0 +1,155 @@
+import { Button, Flex, Heading, Stack } from "@chakra-ui/react";
+import { CreateCollectionOptions } from "arangojs/collection";
+import { Form, Formik } from "formik";
+import React from "react";
+import { mutate } from "swr";
+import * as Yup from "yup";
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader
+} from "../../../components/modal";
+import { getCurrentDB } from "../../../utils/arangoClient";
+import { AddCollectionForm } from "./AddCollectionForm";
+
+const INITIAL_VALUES = {
+  name: "",
+  type: "document" as "document" | "edge",
+  waitForSync: false,
+  numberOfShards: 1,
+  shardKeys: ["_key"],
+  replicationFactor: window.frontendConfig.defaultReplicationFactor,
+  writeConcern: 1,
+  distributeShardsLike: "",
+  isSatellite: false,
+  smartJoinAttribute: ""
+} satisfies Partial<CreateCollectionOptions> & Record<string, any>;
+
+export const AddCollectionModal = ({
+  isOpen,
+  onClose
+}: {
+  isOpen: boolean;
+  onClose: () => void;
+}) => {
+  const handleSubmit = async (values: typeof INITIAL_VALUES) => {
+    const currentDB = getCurrentDB();
+    const {
+      name,
+      type,
+      waitForSync,
+      distributeShardsLike,
+      numberOfShards,
+      replicationFactor,
+      writeConcern,
+      isSatellite,
+      ...extra
+    } = values;
+    const clusterOptions: Partial<CreateCollectionOptions> = window.App
+      .isCluster
+      ? distributeShardsLike
+        ? { distributeShardsLike, ...extra }
+        : {
+            numberOfShards,
+            replicationFactor: (isSatellite
+              ? "satellite"
+              : replicationFactor) as any, // TODO workaround for arangojs 7
+            writeConcern,
+            ...extra
+          }
+      : {};
+    try {
+      await currentDB.collection(name).create({
+        type: type === "edge" ? 3 : 2,
+        waitForSync,
+        ...clusterOptions
+      });
+      window.arangoHelper.arangoNotification(
+        `The collection: ${values.name} was successfully created`
+      );
+      mutate("/collections");
+      onClose();
+    } catch (error: any) {
+      const errorMessage = error?.response?.body?.errorMessage;
+      if (errorMessage) {
+        window.arangoHelper.arangoError("Collection", errorMessage);
+      }
+    }
+  };
+  return (
+    <Modal size="6xl" isOpen={isOpen} onClose={onClose}>
+      <Formik
+        initialValues={INITIAL_VALUES}
+        validationSchema={Yup.object({
+          name: Yup.string().required("Name is required")
```

Comment:
Validations are missing for `name` - may be also based on `window.frontendConfig.extendedNames`. See `window.arangoValidationHelper.getCollectionNameValidations` usage in the old code for more details. We can remove that function and add validations here, since that will be unused now

for eg: 
<img width="690" alt="image" src="https://github.com/arangodb/arangodb/assets/2976363/ba2bb8ad-f070-4eaa-be6a-9d0fe1c9d067">